src/harness/reference_models/move_list_calc/move_list.py

```diff
@@ -0,0 +1,392 @@
+# statements of any third-party software that are legally bundled with
+# the code in compliance with the conditions of those licenses.
+
+#==================================================================================
+# DPA Move List Reference Model.
+# Based on WINNF-TS-0112, Version V1.3.0, 27 September 2017.
+# This version of the code only addresses 'co-channel' frequency range protection
+# in R2-SGN-24.
+
+# The main routine is 'move_list.findMoveList()'
+#==================================================================================
+
+import numpy as np
+from operator import itemgetter
+from shapely.geometry import Point as SPoint
+from shapely.geometry import Polygon as SPolygon
+from collections import namedtuple
+
+# Import WINNF reference models including propagation, geo, and CBSD antenna gain models
+from reference_models.geo import vincenty
+from reference_models.propagation import wf_itm
+from reference_models.antenna import antenna
+
+# Set constant parameters based on requirements in the WINNNF-TS-0112
+CAT_A_NBRHD_DIST = 150          # neighborhood distance for Cat-A CBSD (in km) [R2-SGN-24]
+CAT_B_NBRHD_DIST = 200          # neighborhood distance for Cat-B CBSD (in km) [R2-SGN-24]
+CAT_B_NBRHD_DIST_FOR_HT = 300   # neighborhood distance for Cat-B CBSD (in km) in
+                                # consideration of antenna height [R2-SGN-24]
+CAT_B_LONG_HT = 200             # height of Cat_B CBSD (in meters) in consideration of
+                                # neighborhood distance Cat_B_NBRHD_DIST_FOR_HT [R2-SGN-24]
+PROTECTION_PERCENTILE = 95      # Monte Carlo percentile for protection [R2-SGN-24]
+
+REF_INC_ANTENNA_HT = 50         # reference incumbent radar's antenna height (in meters) [R2-IPM-04]
+REF_INC_ANTENNA_ROT_STEP_SIZE = 1.5 # reference incumbent radar's antenna rotation step size beginning
+                                #  with the main beam boresight at true north (in degrees)
+                                # [R2-IPM-04]
+FREQ_PROP_MODEL = 3625.0        # frequency used in propagation model (in MHz) [R2-SGN-04]
+
+def haat(lat, lon, ht):
+
+    """
+    Convert height above ground level (AGL) to height above average terrain (HAAT) height.
+    TO BE PROVIDED. Just return 'ht' for now.
+    """
+
+    return ht
+
+def findGrantsInsideNeighborhood(reg_request, grant_request, grant_index, constraint,
+                                 exclusion_zone):
+
+    """
+    Identify the Nc CBSD grants in the neighborhood of protection constraint.
+
+    Inputs:
+        reg_request:    a list of CBSD registration requests
+        grant_request:  a list of grant requests
+        grant_index:    an array of grant indices
+        constraint:     protection constraint, a tuple with named fields of
+                        'latitude', 'longitude', 'low_frequency', 'high_frequency'
+        exclusion_zone: a list of locations used to form the exclusion zone
```

Comment:
Good point about passing either an object shaply.geometry.Polygon or lat_lng as an input. I will modify the code and the description accordingly.

Passing a MultiPolygon including only East-Gulf and West coastal exclusion zones to the move_list.py would be ok for co-channel offshore DPAs protection. But I'm afraid that it will get complicated when we incorporate co-channel inland DPAs protection. I would like to use a Polygon for now, but I will keep MultiPolygon option in mind.